src/uart_pl011.rs

```diff
@@ -27,6 +27,10 @@ impl fmt::Write for Pl011 {
     fn write_str(&mut self, s: &str) -> fmt::Result {
         for byte in s.bytes() {
             self.send(byte);
+            // Unix-like OS treats LF as CRLF
+            if byte == b'\n' {
+                self.send(b'\r');
+            }
```

Comment:
Send `\r` just before `\n`.
```suggestion
            // Unix-like OS treats LF as CRLF
            if byte == b'\n' {
                self.send(b'\r');
            }
            self.send(byte);
```

Reply:
Thanks @retrage -, 

Out of curiosity, what's the difference of the different order of LF and CR?

Reply:
CR (Carriage Return) and LF (Line Feed) come from typewriter keys. Although we no longer use typewriters, the combination of the characters is still used for historical reasons.

According to [this Wikipedia article](https://en.wikipedia.org/wiki/Newline#Representation), CR+LF is used by the Windows family, but LF+CR is rarely used.

Reply:
@retrage -, Thank your explanation. Done.

Reply:
Thanks!